### PR TITLE
DSEGOG-17 adding Redux state management

### PR DESCRIPTION
**Pull request by sam-glendenning**
## Description
To allow for complex state management, I am adding Redux. I have based a lot of what I've done on how DataGateway implements Redux, intending to correctly segregate SciGateway state and OperationsGateway state. This also allows for us to start making use of an OperationsGateway settings file, and I've done so by adding in an apiUrl, routes array and pluginHost (none of which are really being used at the moment but will be later on). 

It's likely that, in my attempt to mimic Redux in DataGateway's monorepo structure, I have added something unnecessary or missed something out that we need. Louise will know more about this and can help here.

## Testing instructions
Look over the code and see how things look. I tested that an apiUrl was being read from the settings file by console.logging it in the useRecordsPaginated function, where it is fetched for future use. I've done my best to add tests that ensure the correct actions are firing to store info in state but always good to check this yourself.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes DSEGOG-17

### CODE REVIEW

File: src/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+jest.mock('loglevel');
+jest.mock('axios', () => ({
+  __esModule: true,
+  ...jest.requireActual('axios'),
+  get: jest.fn().mockResolvedValue({}),
+}));
+
+// TODO test index.tsx (currently struggling to mock axios)
+describe.skip('index - fetchSettings', () => {
```
Comment (sam-glendenning) [inline src/index.test.tsx:15]: Would love some help on this one - been struggling to test index.tsx as I'm struggling to mock axios. With this current setup, we get an error of
```
TypeError: _axios.default.get.mockResolvedValue is not a function
```
This makes it impossible to test index.tsx at the moment. Any ideas on how to get this working?

**Comment by sam-glendenning** [comment]
Improving tests on `App.tsx`

File: src/index.test.tsx (hunk 1 above)
Comment (louise-davies) [reply src/index.test.tsx:15]: With axios, I always found it better to use a mock in the `__mocks__` folder as we never actually want original axios functionality in the tests at all, so it should always be mocked by default

**Comment by louise-davies** [comment]
@sam-glendenning I think it's safe to close this PR in favour of the redux-toolkit one?

**Comment by sam-glendenning** [comment]
Yes, I was gonna leave it until the other was merged but it was always gonna happen anyway.

Closing in favour of https://github.com/ral-facilities/operationsgateway/pull/44

**Pull request closed without merging**